### PR TITLE
Oauth setup

### LINKED ISSUE #33: set up Github OAuth

### PR DESCRIPTION
**Pull request by KristyHSC**
closes #33 #47

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -4,6 +4,12 @@ const path = require('path');
 const morgan = require('morgan');
 const api = require('./routes/api');
 
+try {
+  Object.assign(process.env, require('./.env.js'));
```
Comment (leafoflegend) [inline server/app.js]: Consider `.env`

File: package.json [hunk 2]
```diff
@@ -28,6 +28,7 @@
     "@uiw/react-codemirror": "^1.0.19",
     "axios": "^0.18.0",
     "express": "^4.17.0",
+    "gitstar-components": "^1.0.5",
```
Comment (hgranthorner) [inline package.json]: @KristyHSC did you actually use this module anywhere?

File: server/app.js (hunk 1 above)
Comment (hgranthorner) [inline server/app.js]: @KristyHSC Could you please share the .env.js file with us on slack?

Comment (hgranthorner) [reply server/app.js]: Also, could you add the .env.js file to the .gitignore? I don't see a change for that file in this pr

File: client/components/Login.js [hunk 3]
```diff
@@ -14,6 +14,10 @@ const LoginClass = props => {
       .then(() => props.history.push('/'));
   };
 
+  const GithubOauth = () => {
```
Comment (hgranthorner) [inline client/components/Login.js]: Let's make sure we stick with camel cased function names: githubOAuth

File: server/app.js (hunk 1 above)
Comment (KristyHSC) [reply server/app.js]: I uninstall the gitstar-components, and add .env.js to .gitignore. :)

File: client/components/Login.js [hunk 4]
```diff
@@ -33,6 +37,10 @@ const LoginClass = props => {
         />
         <button type="submit">Login</button>
       </form>
+      <button type="button" onClick={githubOauth}>
+        {' '}
```
Comment (chem2016) [inline client/components/Login.js]: remove 41 {''} and the next one

**Pull request merged**